app/Http/Controllers/UserController.php

```diff
@@ -57,26 +56,38 @@ public function login(Request $request): JsonResponse
             $request->session()->regenerate();
             return new JsonResponse();
         }
-        throw ExceptionFactory::unauthorised();
+        Auth::logout();
+        return ExceptionFactory::badCredentials()->render();
```

Comment:
skoro nie uda się zalogować i wywali exceptiona to po co u góry te `Auth::logout()`?

Reply:
jak jestem zalogowany jako A i próbuję się zalogować jako B, ale podałem błędne dane, to lepiej mnie wylogować, żeby była jasna komunikacja. Niby badCredentials, ale w sumie jestem zalogowany